pkg/controller/masteruserrecord/sync.go

```diff
@@ -61,6 +87,55 @@ func (s *Synchronizer) synchronizeStatus() error {
 	return nil
 }
 
+// withClusterDetails returns the given user account status with additional information about
+// the target cluster such as API endpoint and Console URL if not set yet
+func (s *Synchronizer) withClusterDetails(status toolchainv1alpha1.UserAccountStatusEmbedded) (toolchainv1alpha1.UserAccountStatusEmbedded, error) {
+	if status.Cluster.Name != "" && (status.Cluster.APIEndpoint == "" || status.Cluster.ConsoleURL == "") {
```

Comment:
I'm just curious: when the `status.Cluster.Name` could be empty?

Reply:
When there is no status (yet). In this case the status struct is an empty struct with empty strings by default.